app/src/main/java/se/rebeccazadig/bokholken/adverts/AdvertsFragment.kt

```diff
@@ -1,29 +1,30 @@
 package se.rebeccazadig.bokholken.adverts
 
 import android.os.Bundle
+import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.appcompat.widget.SearchView
 import androidx.fragment.app.Fragment
-import androidx.fragment.app.viewModels
+import androidx.lifecycle.ViewModelProvider
 import androidx.navigation.fragment.findNavController
 import androidx.recyclerview.widget.LinearLayoutManager
-import se.rebeccazadig.bokholken.R
-import se.rebeccazadig.bokholken.data.Advert
 import se.rebeccazadig.bokholken.databinding.FragmentAdvertsBinding
+import se.rebeccazadig.bokholken.models.Advert
 
 class AdvertsFragment : Fragment() {
 
     private var _binding: FragmentAdvertsBinding? = null
     private val binding get() = _binding!!
 
-    private val viewModel: AdvertViewModel by viewModels()
+    private val viewModel by lazy {
```

Comment:
out of curiosity, Why not use by?

Reply:
When I change it from ViewModel to AndroidViewmodel, I got an error but  I needed to have ViewModelProvider. But now by viewModels() work. So I changed it back